packages/odyssey-react-mui/src/renderOktaUiShell.tsx

```diff
@@ -0,0 +1,52 @@
+/*!
+ * Copyright (c) 2024-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { bufferUntil } from "./bufferUntil";
+import { createMessageBus } from "./createMessageBus";
+import { OktaUiShell, type OktaUiShellComponentProps } from "./OktaUiShell";
+import { renderReactInWebComponent } from "./renderReactInWebComponent";
+
+export const renderOktaUiShell = ({
+  contentElementId,
+  rootElement,
+}: {
+  contentElementId?: string;
+  rootElement: HTMLElement;
+}) => {
+  const { publish: publishPropChanges, subscribe: subscribeToPropChanges } =
+    createMessageBus<OktaUiShellComponentProps>();
+
+  const {
+    publish: publishSubscriptionCreated,
+    subscribe: subscribeToReactAppSubscribed,
+  } = createMessageBus();
+
+  const publishAfterReactAppReadyForProps = bufferUntil({
+    publish: publishPropChanges,
+    subscribe: subscribeToReactAppSubscribed,
+  });
```

Comment:
When trying to publish into a React app, you need to wait for the React app to be rendered, but there's no way to know that except inside the React app itself.

The web component also adds lag, so that creates a problem that needed a "tell me when you're done" solution. This method won't have timing issues.

---

`bufferUntil` queues up the last `publish` call and waits till the React app is subscribed for props. Once that happens, it stops buffering.